scss/_input-group.scss

```diff
@@ -179,6 +179,7 @@
 .input-group > .input-group-append:not(:last-child) > .btn,
 .input-group > .input-group-append:not(:last-child) > .input-group-text,
 .input-group > .input-group-append:last-child > .btn:not(:last-child):not(.dropdown-toggle),
+.input-group > .input-group-append:last-child > .dropdown-toggle:nth-last-child(n + 3),
```

Comment:
`(n + 3)` stands for third last or more. This way the radius is removed only if the `.dropdown-toggle` is not the second last element ( and `.dropdown-menu` is the last). Couldn't combine this with L181, because complex selectors for `:not()` are not well supported.

Reply:
This feels like it'd break very easily—folks introduce custom markup in all the places we wouldn't expect. 😬